lib/app_web/live/app_live.ex

```diff
@@ -108,15 +116,246 @@ defmodule AppWeb.AppLive do
+              chs_stop = chs.data.stop
+
+              # The condition needs to FAIL (StartA <= EndB) and (EndA >= StartB)
+              # so no timer overlaps one another
+              compareStartAEndB = NaiveDateTime.compare(start, chs_stop)
+              compareEndAStartB = NaiveDateTime.compare(stop, chs_start)
+
+              if(
+                (compareStartAEndB == :lt || compareStartAEndB == :eq) &&
+                  (compareEndAStartB == :gt || compareEndAStartB == :eq)
+              ) do
+                throw(:overlap)
+              end
+            end
+
+            Timer.update_timer(%{id: id, start: start, stop: stop})
+            {:noreply, assign(socket, editing: nil, editing_timers: [])}
+          catch
+            :overlap ->
+              updated_changeset_timers_list =
+                error_timer_changeset(
+                  timer_changeset_list,
+                  changeset_obj,
+                  index,
+                  :id,
+                  "This timer interval overlaps with other timers. Make sure all the timers are correct and don't overlap with each other"
+                )
+
+              {:noreply,
+               assign(socket, editing_timers: updated_changeset_timers_list)}
+          end
+
+        :eq ->
+          updated_changeset_timers_list =
+            error_timer_changeset(
+              timer_changeset_list,
+              changeset_obj,
+              index,
+              :id,
+              "Start or stop are equal."
+            )
+
+          {:noreply,
+           assign(socket, editing_timers: updated_changeset_timers_list)}
+
+        :gt ->
+          updated_changeset_timers_list =
+            error_timer_changeset(
+              timer_changeset_list,
+              changeset_obj,
+              index,
+              :id,
+              "Start is newer that stop."
+            )
+
+          {:noreply,
+           assign(socket, editing_timers: updated_changeset_timers_list)}
+      end
+    rescue
+      _e ->
+        updated_changeset_timers_list =
+          error_timer_changeset(
+            timer_changeset_list,
+            changeset_obj,
+            index,
+            :id,
+            "Date format invalid on either start or stop."
+          )
+
+        {:noreply,
+         assign(socket, editing_timers: updated_changeset_timers_list)}
+    end
+  end
+
+  # Errors a specific changeset from a list of changesets and returns the updated list of changesets.
+  # You should pass a:
+  # - `timer_changeset_list: list of timer changesets to be updated
+  # - `changeset_to_error`: changeset object that you want to error out
+  # - `changeset_index`: changeset object index inside the list of timer changesets
+  # - `changeset_error_key`: atom key of the changeset object you want to associate the error message
+  # - `changeset_error_message`: the string message to error the changeset key with.
+  @doc false
+  defp error_timer_changeset(
```

Comment:
This `timer_changeset` error handling code should _definitely_ not be in the `lib/app_web/live/app_live.ex` file.